crates/rtmp/src/session/server_session.rs

```diff
@@ -193,206 +200,188 @@ impl<S: tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin> Session<S> {
                 .io
                 .read_buf(&mut self.read_buf)
                 .with_timeout(Duration::from_millis(2500))
-                .await??;
+                .await
+                .map_err(SessionError::Timeout)??;
 
             if n == 0 {
                 return Ok(false);
             }
+
+            let n = n.try_into().unwrap_or(u32::MAX);
```

Comment:
this is ALWAYS less then u32::MAX since this is the number of bytes read and the read_buf is bound.